Formula/sapling.rb

```diff
@@ -0,0 +1,33 @@
+class Sapling < Formula
+  desc "Source control client"
+  homepage "https://sapling-scm.com"
+  url "https://github.com/facebook/sapling/archive/refs/tags/0.1.20221118-210929-cfbb68aa.tar.gz"
+  version "0.1.20221118-210929-cfbb68aa"
+  sha256 "51ce336f1eb382e591a7384d0292cb41fafbbd92cc701ee74b50099beece65e4"
+  license "GPL-2.0-or-later"
+  head "https://github.com/facebook/sapling.git", branch: "main"
+
+  depends_on "cmake" => :build
+  depends_on "rust" => :build
+  depends_on "yarn" => :build
+  depends_on "gh"
+  depends_on "node"
+  depends_on "openssl@1.1"
+  depends_on "python@3.8"
+
+  def install
+    ENV["OPENSSL_DIR"] = Formula["openssl@1.1"].opt_prefix
+    ENV["PYTHON_SYS_EXECUTABLE"] = Formula["python@3.8"].opt_prefix/"bin/python3.8"
+    ENV["PYTHON"] = Formula["python@3.8"].opt_prefix/"bin/python3.8"
+    ENV["PYTHON3"] = Formula["python@3.8"].opt_prefix/"bin/python3.8"
+    ENV["SAPLING_VERSION"] = version.to_s
+
+    cd "eden/scm" do
+      system "make", "PREFIX=#{prefix}", "install-oss"
+    end
+  end
+
+  test do
+    assert_equal("Sapling #{version}", shell_output("#{bin}/sl --version").chomp)
```

Comment:
We need a better test in here to test the actual functionality of the commands

Reply:
Sounds ok to me. Just before I go ahead and actually start writing some more complex test, do you have anything in mind for testing it? Some other formulas seem to just test that the version command works:

https://github.com/Homebrew/homebrew-core/blob/master/Formula/watchman.rb#L74

Reply:
Added a slightly more complex test.